etk-asm/src/asm.rs

```diff
@@ -353,39 +360,59 @@ impl Assembler {
     }
 
     /// Feed a single instruction into the `Assembler`.
-    ///
-    /// Returns the number of bytes that can be collected with [`Assembler::take`]
-    pub fn push<O>(&mut self, rop: O) -> Result<usize, Error>
+    fn push<O>(&mut self, rop: O) -> Result<usize, Error>
     where
         O: Into<RawOp>,
     {
         let rop = rop.into();
 
-        self.declare_content(&rop)?;
+        self.declare_label(&rop)?;
 
-        // Expand instruction macros immediately. We do this here because it's the same process
-        // regardless if we `push_read` or `push_pending` -- in fact, `expand_macro` pushes each op
-        // individually which calls the correct unchecked push.
+        // Expand instruction macros immediately.
         if let RawOp::Op(AbstractOp::Macro(ref m)) = rop {
             self.expand_macro(&m.name, &m.parameters)?;
-            return Ok(self.ready.len());
+            return Ok(self.concrete_len);
         }
 
-        self.push_unchecked(rop)?;
-        Ok(self.ready.len())
+        self.push_rawop(rop)?;
+        Ok(self.concrete_len)
     }
```

Comment:
Is it possible to combine `push` with `push_rawop`? Like move `self.expand_macro(&m.name, &m.parameters)?;` (line 373) to line 424?